src/btree/bt_read.c

```diff
@@ -125,17 +128,258 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+        case WT_PAGE_COL_VAR:
+            p = las_key.data;
+            WT_ERR(__wt_vunpack_uint(&p, 0, &recno));
+            WT_ERR(__col_instantiate(session, recno, ref, &cbt, upd));
+            break;
+        case WT_PAGE_ROW_LEAF:
+            WT_ERR(__row_instantiate(session, &las_key, ref, &cbt, upd));
+            break;
+        default:
+            WT_ERR(__wt_illegal_value(session, page->type));
+        }
+
+        /* Remove the prepared record from LAS once the page is instantiated successfully */
+        if (!birthmark_record && upd->prepare_state == WT_PREPARE_INPROGRESS) {
+            /* Extend the buffer if needed */
+            WT_ERR(__wt_buf_extend(session, las_prepares,
+              (las_prepare_cnt + 1) * sizeof(struct las_page_prepared_updates)));
+            las_preparep = (struct las_page_prepared_updates *)las_prepares->mem + las_prepare_cnt;
+            WT_CLEAR(las_preparep->key);
+            WT_ERR(__wt_buf_set(session, &las_preparep->key, las_key.data, las_key.size));
+            las_preparep->timestamp = las_timestamp;
+            las_preparep->txnid = las_txnid;
+            las_prepare_cnt++;
+        }
+
+        if (birthmark_record) {
+            birthmarkp->instantiated = true;
+            birthmark_record = false;
+        }
+
+        upd = NULL;
+    }
+
+    __wt_readunlock(session, &cache->las_sweepwalk_lock);
+    locked = false;
+    WT_ERR_NOTFOUND_OK(ret);
+
+    __wt_verbose(session, WT_VERB_LOOKASIDE_ACTIVITY,
+      "Page is instantiated with LAS content for btree ID %" PRIu32
+      " with number of keys: %" PRId32,
+      las_btree_id, instantiated_cnt);
+
+    /* Instantiate the remaining birthmark entries */
+    WT_ERR(__instantiate_birthmarks(session, ref));
+
+    /*
+     * Now the page is successfully instantiated. It is unlikely that removing of prepared updates
+     * to fail.
+     */
+    for (i = 0, las_preparep = las_prepares->mem; i < las_prepare_cnt; i++, las_preparep++) {
+        cursor->set_key(
+          cursor, las_btree_id, &las_preparep->key, las_preparep->timestamp, las_preparep->txnid);
+        WT_ERR(cursor->remove(cursor));
+    }
+    /* Discard the cursor. */
+    WT_ERR(__wt_las_cursor_close(session, &cursor, session_flags));
+
+    __wt_buf_free(session, &ref->page_las->max_las_key);
+    __wt_buf_free(session, &ref->page_las->min_las_key);
+
+err:
+    if (las_prepare_cnt != 0)
+        for (i = 0, las_preparep = las_prepares->mem; i < las_prepare_cnt; i++, las_preparep++)
+            __wt_buf_free(session, &las_preparep->key);
+
+    __wt_scr_free(session, &las_prepares);
+    for (birthmark_cnt = 0, birthmarkp = ref->page_las->birthmarks;
+         birthmark_cnt < ref->page_las->birthmarks_cnt; birthmark_cnt++, birthmarkp++)
+        birthmarkp->instantiated = false;
+
+    if (locked)
+        __wt_readunlock(session, &cache->las_sweepwalk_lock);
+    WT_TRET(__wt_las_cursor_close(session, &cursor, session_flags));
```

Comment:
In the non-error case we are closing this cursor twice.

Reply:
Fixed.